generic-extras/src/test/scala/io/circe/generic/extras/EnumerationSemiautoDerivedSuite.scala

```diff
@@ -46,9 +45,15 @@ class EnumerationSemiautoDerivedSuite extends CirceSuite {
   )
 
   test("deriveEnumerationDecoder should not compile on an ADT with case classes") {
-    implicit val config: Configuration = Configuration.default
-    val _ = config
-    illTyped("deriveEnumerationDecoder[ExtendedCardinalDirection]")
+    assert(
+      compileErrors(
+        """
+        implicit val config: Configuration = Configuration.default
+        val _ = config
+        deriveEnumerationDecoder[ExtendedCardinalDirection]
+        """
+      ).nonEmpty
```

Comment:
I think we need some assertions here that the errors we are expecting are reasonable.  For instance, in the future this will be a nonsense test if `ExtendedCardinalDirection` gets deleted or renamed, or some similar change happens that has nothing to do with deriving enumeration decoders.

Reply:
Makes sense.
On the other hand, the test could become a little bit more volatile, as it'll be susceptible to changes in compiler error messages, but maybe we can live with that.